src/caching/Arweave/ArweaveClient.ts

```diff
@@ -225,21 +233,45 @@ export class ArweaveClient {
     return this.client.wallets.jwkToAddress(this.arweaveJWT);
   }
 
+  private async _retryRequest(request: () => Promise<unknown>, retryCount: number): Promise<unknown> {
```

Comment:
ooc, can this work as a generic so we don't need the type assertion to `BigNumber` down on line 275?

Reply:
690d87977daee0aa83f602ac1421e349da8850df 


good idea